src/core/constraint_template.jl

```diff
@@ -61,6 +61,39 @@ function constraint_resistor_weymouth(gm::AbstractGasModel, k; n::Int=gm.cnw)
 end
 
 
+###############################################################################################
+# Templates for constraints associated with prvs
+###############################################################################################
+
+"Template: Constraint on mass flow across a prv"
+function constraint_prv_mass_flow(gm::AbstractGasModel, k; n::Int=gm.cnw)
+    prv              = ref(gm, n, :prv, k)
+    f_min            = prv["flow_min"]
+    f_max            = prv["flow_max"]
+    is_bidirectional = get(prv, "is_bidirectional", 1)
+    flow_direction   = get(prv, "flow_direction", 0)
+
+    if is_bidirectional == 0 || flow_direction == 1

```

Comment:
Take a look at how I am computing f_min and f_max now.  I am now doing this in _calc_component_max_flow... Basically, what I am storing in the ref is the tighted f_min and f_max bounds.  I found this logic was being done over and over again, so it made more sense to go ahead and do it in the calc function.